sparta/sparta/statistics/BasicHistogram.hpp

```diff
@@ -0,0 +1,63 @@
+// <BasicHistogram.h> -*- C++ -*-
+
+/**
+ * \file BasicHistogram.hpp
+ * \brief A simple histogram with programmable ranges, using sparta::Counters
+ */
+#pragma once
+
+#include "sparta/statistics/Counter.hpp"
+
+/// Histogram with programmable buckets
+class BasicHistogram
```

Comment:
I think this class could be extended easily to include different bucket types (not just `vector of ints`  as well as a user defined comparator function that can be passed to the `lower_bound` method.